src/OrchardCore.Modules/OrchardCore.Lucene/GraphQL/LuceneQueryFieldTypeProvider.cs

```diff
@@ -77,38 +79,45 @@ public async Task BuildAsync(ISchema schema)
             }
         }
 
-        private FieldType BuildSchemaBasedFieldType(LuceneQuery query, JToken schema)
+        private FieldType BuildSchemaBasedFieldType(LuceneQuery query, JToken querySchema)
         {
+            var properties = querySchema["properties"];
+            if (properties == null)
+            {
+                return null;
+            }
+
             var typetype = new ObjectGraphType<JObject>
             {
                 Name = query.Name
             };
 
-            var properties = schema["Properties"];
-
-            foreach (var child in properties.Children())
+            foreach (JProperty child in properties.Children())
             {
-                var name = ((JProperty)child).Name;
+                var name = child.Name;
                 var nameLower = name.Replace('.', '_');
-                var type = child["type"].ToString();
+                var type = child.Value["type"].ToString();
+                var description = child.Value["description"]?.ToString();
 
-                if (type == "String")
+                if (type == "string")
```

Comment:
Why do you change the capitalization here from `String` to `string` ?

Would it be better just to do a `String.Equals("string", type, StringComparison.OrdinalIgnoreCase)` case insensitive comparison? 

To be non breaking on existing queries?

Reply:
I changed it so it's according to [official schema](https://json-schema.org/learn/miscellaneous-examples.html). And regarding to 'non breaking change' this commit is actually to fix bug because I wasn't even able to run this feature, i.e. it didn't work at all

Reply:
ok makes sense.

I updated the docs a little, so mkdocs will flow them properly.

Good job fixing this, I feel like it's been 'fixed' two or three times in the last year or two.

I agree about the logging it would be a nice improvement.

Do you want to do it on this pr, or prefer to add another time?

Reply:
Actually not sure what to do. Maybe it's more efficient to remove null checking on properties and typetype variable. And let `properties.Children()` and `schema.Query.Fields.OfType<ContentItemsFieldType>().First(x => x.Name == contentType)` throw errors (and let buildAsync to catch it) because those cases should happen rarely and there is no reason to have checking for null that will happen almost never.

And secondly what're the rules around here about unit testing are you avoiding them or there just isn't enough time.